src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -414,12 +420,51 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		return this._defaultLanguageInfo;
 	}
 
+	public get language(): string {
+		return this._language;
+	}
+
 	private updateLanguageInfo(info: nb.ILanguageInfo) {
 		if (info) {
 			this._defaultLanguageInfo = info;
+			this.trySetLanguageFromLangInfo();
 		}
 	}
 
+	private trySetLanguageFromLangInfo() {
```

Comment:
some tests so that we don't regress this again in the future would be fantastic 😄